packages/core/src/components/search/Filter/FilterDeliveryOption.tsx

```diff
@@ -0,0 +1,76 @@
+import { Button as UIButton } from '@faststore/ui'
+import { sessionStore } from 'src/sdk/session'
+import { textToTitleCase } from 'src/utils/utilities'
+
+interface DeliveryCustomLabels {
+  delivery?: string
+  pickupInPoint?: string
+  pickupNearby?: string
+  pickupAll?: string
+}
+
+interface FacetValue {
+  value: string
+  label: string
+  selected: boolean
+  quantity: number
+}
+
+interface FilterDeliveryOptionProps {
+  item: FacetValue
+  deliveryCustomLabels: DeliveryCustomLabels
+}
+
+export function FilterDeliveryOption({
+  item,
+  deliveryCustomLabels,
+}: FilterDeliveryOptionProps) {
+  const { city, postalCode } = sessionStore.read()
+  const location = city ? `${textToTitleCase(city)}, ${postalCode}` : postalCode
+
+  const mapShippingLabel: Record<string, string> = {
+    delivery: deliveryCustomLabels?.delivery ?? 'Deliver to',
+    'pickup-in-point': deliveryCustomLabels?.pickupInPoint ?? 'Pickup at',
+    'pickup-nearby': deliveryCustomLabels?.pickupNearby ?? 'Pickup Nearby',
+    'pickup-all': deliveryCustomLabels?.pickupAll ?? 'Pickup Anywhere',
+  }
+
+  if (item.value === 'delivery') {
+    return (
+      <>
+        {mapShippingLabel[item.value]}
+        <UIButton
+          data-fs-filter-list-item-button
+          size="small"
+          onClick={() => {
+            // TODO: open edit local slideOver
+            window.alert('Open Modal')
+          }}
+        >
+          {location}
+        </UIButton>
+      </>
+    )
+  }
+
+  if (item.value === 'pickup-in-point') {
+    return (
+      <>
+        {mapShippingLabel[item.value]}
+        <UIButton
+          data-fs-filter-list-item-button
+          size="small"
+          onClick={() => {
+            // TODO: open edit local slideOver
+            window.alert('Open Modal')
+          }}
+        >
+          Robson St
```

Comment:
I couldn't visualize this case, because no zipCode that I've tried had this option 😞 

remind to change this in the next task.

Reply:
I believe this won't come automatically from IS, we can get the closest pickup point from the defined postal code using the [API that lists pickup points](https://docs.google.com/document/d/1FTBcTj7pPv9acl88mxXho2nwDSEPHw4BUKixTqKgKKs/edit?tab=t.0#heading=h.enq5trfo8p8).

And yes, we should do this integration in the next task!